### PR TITLE
add "create sol github action" to the FE menu

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts [hunk 1]
```diff
@@ -533,6 +533,33 @@ export const createNewBranch = async (branch: string) => {
...
+  on: [push]
+
+  jobs:
+    run_sol_contracts_job:
+      runs-on: ubuntu-latest
+      name: A job to run solidity unit tests on github actions CI
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v2
+        - name: Environment Setup
```
Comment (ioedeveloper) [inline libs/remix-ui/workspace/src/lib/actions/workspace.ts:548]: This step is not so necessary for a basic setup.

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx [hunk 2]
```diff
@@ -531,6 +535,40 @@ export function Workspace () {
...
+        onClick={(e) => {
+          e.stopPropagation()
+          addGithubAction()
+          _paq.push(['trackEvent', 'fileExplorer', 'workspaceMenu', 'addSolidityTesting'])
+          hideIconsMenu(!showIconsMenu)
+        }}
+        className='fab fa-github pl-2'
+      >
+      </span>
+      <span className="pl-3">{'Add Soldity Unit test github action'}</span>
```
Comment (ioedeveloper) [inline libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:569]: Can we find a shorter title? It takes up so much space on the UI.

Comment (ioedeveloper) [reply libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:569]: Also, i'm not sure if this option should be only visible for git workspaces. What do you think?

**Comment by ioedeveloper** [comment]
We can also add a context menu option for this for files ending with `_test.sol`

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts (hunk 1 above)
Comment (yann300) [reply libs/remix-ui/workspace/src/lib/actions/workspace.ts:548]: ok I removed it

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx (hunk 2 above)
Comment (yann300) [reply libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:569]: I would make it visible for all the workspace I think.. it gives a bit more visibilty to this feature an encourage to commit workspace to github..

**Review by joeizang — APPROVED** [review]
Maybe we can add syntax highlighting for yaml files to the monaco editor later on?

**Pull request merged**